### PR TITLE
Run framework tests on OCM PRs

### PR DESCRIPTION
**Pull request by zyjjay**
ref: https://issues.redhat.com/browse/ACM-4811

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -187,6 +198,13 @@ kind-controller-all-kubeconfigs:
 	CLUSTER_NAME=$(HUB_CLUSTER_NAME) KIND_NAME=$(KIND_HUB_NAME) KIND_CLUSTER_NAME=$(KIND_HUB_CLUSTER_NAME) $(MAKE) kind-controller-kubeconfig
 	yq e '.clusters[0].cluster.server = "https://$(KIND_HUB_NAME)-control-plane:6443"' $(HUB_CONFIG) > $(HUB_CONFIG_INTERNAL)
 
+.PHONY: kind-deploy-controller-dev-addon
+kind-deploy-controller-dev-addon:
+	@echo Hosted mode test
+	kind load docker-image $(REGISTRY)/$(IMG):$(TAG) --name $(KIND_NAME)
+	kubectl annotate -n $(subst -hosted,,$(KIND_NAMESPACE)) --overwrite managedclusteraddon governance-policy-framework\
+		addon.open-cluster-management.io/values='{"global":{"imagePullPolicy": "Never", "imageOverrides":{"governance_policy_framework_addon": "$(REGISTRY)/$(IMG):$(TAG)"}}}'
+
```
Comment (yiraeChristineKim) [inline Makefile:207]:
```
apiVersion: addon.open-cluster-management.io/v1alpha1
kind: ManagedClusterAddOn
metadata:
  name: governance-policy-framework
  annotations:
    addon.open-cluster-management.io/hosting-cluster-name: cluster1
    addon.open-cluster-management.io/values: '{"global":{"imageOverrides":{"governance-policy-framework":"quay.io/stolostron/governance-policy-framework-addon:imagetag"}}}'
spec:
  installNamespace: cluster2-hosted
```

Comment (yiraeChristineKim) [reply Makefile:207]: so 
```suggestion
.PHONY: kind-deploy-controller-dev-addon
kind-deploy-controller-dev-addon:
	@echo Hosted mode test
	kind load docker-image $(REGISTRY)/$(IMG):$(TAG) --name $(KIND_NAME)
	kubectl annotate -n $(subst -hosted,,$(KIND_NAMESPACE)) --overwrite managedclusteraddon governance-policy-framework\
		addon.open-cluster-management.io/values='{"global":{"imagePullPolicy": "Never", "imageOverrides":{"governance_policy_framework": "$(REGISTRY)/$(IMG):$(TAG)"}}}'
```

Comment (yiraeChristineKim) [reply Makefile:207]: oh why? ` imagePullPolicy": "Never"`

File: Makefile [hunk 2]
```diff
@@ -187,6 +198,13 @@ kind-controller-all-kubeconfigs:
 	CLUSTER_NAME=$(HUB_CLUSTER_NAME) KIND_NAME=$(KIND_HUB_NAME) KIND_CLUSTER_NAME=$(KIND_HUB_CLUSTER_NAME) $(MAKE) kind-controller-kubeconfig
 	yq e '.clusters[0].cluster.server = "https://$(KIND_HUB_NAME)-control-plane:6443"' $(HUB_CONFIG) > $(HUB_CONFIG_INTERNAL)
 
+.PHONY: kind-deploy-controller-dev-addon
+kind-deploy-controller-dev-addon:
+	@echo Hosted mode test
+	kind load docker-image $(REGISTRY)/$(IMG):$(TAG) --name $(KIND_NAME)
+	kubectl annotate -n $(subst -hosted,,$(KIND_NAMESPACE)) --overwrite managedclusteraddon governance-policy-framework\
+		addon.open-cluster-management.io/values='{"global":{"imagePullPolicy": "Never", "imageOverrides":{"governance_policy_framework_addon": "$(REGISTRY)/$(IMG):$(TAG)"}}}'
```
Comment (yiraeChristineKim) [inline Makefile:206]:
```suggestion
.PHONY: kind-deploy-controller-dev-addon
kind-deploy-controller-dev-addon:
	@echo Hosted mode test
	kind load docker-image $(REGISTRY)/$(IMG):$(TAG) --name $(KIND_NAME)
	kubectl annotate -n $(subst -hosted,,$(KIND_NAMESPACE)) --overwrite managedclusteraddon governance-policy-framework\
		addon.open-cluster-management.io/values='{"global":{"imagePullPolicy": "Never", "imageOverrides":{"governance_policy_framework": "$(REGISTRY)/$(IMG):$(TAG)"}}}'
```

File: .github/workflows/framework-kind.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+name: Framework
+
+on:
+  pull_request:
+    branches:
+      - main
+      - release-[0-9]+.[0-9]+
```
Comment (JustinKuli) [inline .github/workflows/framework-kind.yml:7]: We don't have "release" branches in this repo, but we do have tags... can this workflow be run when we push a tag like vX.Y.Z?

File: Makefile (hunk 2 above)
Comment (zyjjay) [reply Makefile:206]: I cherry-picked this commit from stolostron in order to fix the hosted mode tests. Is there a reason we would use `governance_policy_framework` over the `framework_addon`?

File: .github/workflows/framework-kind.yml (hunk 3 above)
Comment (zyjjay) [reply .github/workflows/framework-kind.yml:7]: I'm able to enable the workflows on `push`es to the repo. I'm thinking something like `v**` would work?

File: Makefile (hunk 2 above)
Comment (yiraeChristineKim) [reply Makefile:206]: Oh I checked addon-controller governance_policy_framework_addon should be right I should change framework-test! Good!

**Review by yiraeChristineKim — APPROVED** [review]

**Pull request merged**